### PR TITLE
fix(dag_warning): rename argument error_type as warning_type

### PR DESCRIPTION
**Pull request by Lee-W**
## Why
The argument error_type in `DagWarning.__init__` is later set as the value of `DagWarning.warning_type`. Not sure whether it's expected

## What
Rename argument error_type as warning_type in `DagWarning.__init__`

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by phanikumv — APPROVED** [review]

**Review by ephraimbuddy — APPROVED** [review]

File: .github/workflows/news-fragment.yml [hunk 1]
```diff
@@ -38,6 +38,7 @@ jobs:
 
       - name: Check news fragment
         run: >
+          python -m pip install --upgrade uv &&
```
Comment (ashb) [inline .github/workflows/news-fragment.yml]: This seems like an unrelated change?

File: newsfragments/43877.significant.rst
Comment (ashb) [inline newsfragments/43877.significant.rst:1]: I don't think we need this fragment -- this feels like an internal class only so this feels like noise to users.

File: .github/workflows/news-fragment.yml (hunk 1 above)
Comment (Lee-W) [reply .github/workflows/news-fragment.yml]: let me rebase it. It's merged in another PR.

File: newsfragments/43877.significant.rst
Comment (Lee-W) [reply newsfragments/43877.significant.rst:1]: I'm kinda neutral on this one 🤔 I'll keep it for one or two hours and see whether anyone else have any thoughts. if not, I'll just remove it

File: .github/workflows/news-fragment.yml (hunk 1 above)
Comment (Lee-W) [reply .github/workflows/news-fragment.yml]: removed

**Review by amoghrajesh — APPROVED** [review]

File: newsfragments/43877.significant.rst
Comment (Lee-W) [reply newsfragments/43877.significant.rst:1]: Removed
